### PR TITLE
HPCC-30942 Fix SOAPCALL handling of lowercase HTTP headers in response

### PR DESCRIPTION
**Pull request by afishbeck**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [x] Send notifications about my Pull Request position in Smoketest queue.
- [x] Test my draft Pull Request.

### CODE REVIEW

File: system/jlib/jstring.cpp [hunk 1]
```diff
@@ -2842,3 +2842,63 @@ void processOptionString(const char * options, optionCallback callback)
...
+ * Hereby donated to public domain.
+ *
+ * Returns:  char *pointer if needle is found in haystack, otherwise NULL.
+ *
+ * Rev History:  11/30/23                    JLibify
+ *               01/20/05  Joachim Nilsson   Cleanups
+ *               07/04/95  Bob Stout         ANSI-fy
+ *               02/03/94  Fred Cole         Original
+ */
+
```
Comment (afishbeck) [inline system/jlib/jstring.cpp:2866]: @ghalliday strcasestr() didn't compile on windows-22.  Tweaked a public domain version of stristr().  Let me know if you'd prefer another approach.

**Review by ghalliday — COMMENTED** [review]
@afishbeck I'm not sure what I think.  It is probably the minimal change to fix the bug, but I had been wondering about the implementation of the current functions.  (Only spotted because I have been looking at this code recently.)
The stristr function is a good idea, but I think the implementation could be improved.  It could also do with some unit tests (along with everything else in the system!)

So we could take it, and then improve/fix the functions as a separate PR, or we could improve them and fix this at the same time.
What are your thoughts?

File: system/jlib/jstring.cpp [hunk 2]
```diff
@@ -2842,3 +2842,63 @@ void processOptionString(const char * options, optionCallback callback)
...
+const char * stristr (const char *haystack, const char *needle)
+{
+    if (isEmptyString(haystack) || isEmptyString(needle))
+        return nullptr;
+
+    const char *pptr  = needle;     /* Pattern to search for    */
+    const char *start = haystack;   /* Start with a bowl of hay */
+    const char *sptr;               /* Substring pointer        */
+    int   slen  = strlen(haystack); /* Total size of haystack   */
+    int   plen  = strlen(needle);   /* Length of our needle     */
```
[the remark this answers is not in the document]
Comment (afishbeck) [reply system/jlib/jstring.cpp:2876]: Yes, I think the p was for pattern.  But confusing, and really a prefix of "p" for anything but pointer can be confusing.

File: system/jlib/jstring.cpp [hunk 3]
```diff
@@ -2842,3 +2842,63 @@ void processOptionString(const char * options, optionCallback callback)
...
+        {
+            start++;
+            slen--;
+            /* if pattern longer than string */
+            if (slen < plen)
+                return nullptr;
+        }
+
+        sptr = start;
+        pptr = (char *) needle;
```
[the remark this answers is not in the document]
Comment (afishbeck) [reply system/jlib/jstring.cpp:2892]: Right.  Originally they weren't const strings but I changed it and forgot this one.

**Comment by afishbeck** [comment]
> @afishbeck I'm not sure what I think. It is probably the minimal change to fix the bug, but I had been wondering about the implementation of the current functions. (Only spotted because I have been looking at this code recently.) The stristr function is a good idea, but I think the implementation could be improved. It could also do with some unit tests (along with everything else in the system!)
> 
> So we could take it, and then improve/fix the functions as a separate PR, or we could improve them and fix this at the same time. What are your thoughts?

@ghalliday I definitely noticed the code could use improvement.  I was trying to minimize the fix to support HAProxy 2.0 thinking the change would likely go into an older branch (not sure how far back).  If it were targeting latest I would definitely recommend improving the code right away for this fix I'm not sure.  I can add the low risk improvements and open a jira for further refactoring?

**Comment by ghalliday** [comment]
I will merge as-is, but have created a jira to go back and clean it up.

**Pull request merged**